packages/components/src/local/organisms/setting-channels/helpers/generateRowItemsCollab.ts

```diff
@@ -0,0 +1,93 @@
+import { CollaborativePermission } from '@serge/config'
+import { ForceData, ParticipantCollab } from '@serge/custom-types'
+import { EDITABLE_SELECT_ITEM, EDITABLE_SWITCH_ITEM, Item, Option } from '../../../molecules/editable-row'
+import { Participant } from '../types/props'
+
+export default (forces: Array<ForceData>, nextParticipant: Participant): Array<Item> => {
+  // by default selected force
+  let forceSelected: Array<number> = [0]
+  // init empty roles array
+  let roleOptions: Array<Option> = []
+
+  // additional table cols for RFI participation toggles
+  const additionalFields: Item[] = []
+
+  if (nextParticipant.forceUniqid) {
+    // get selected force for current row
+    const forceIndex = forces.findIndex(force => force.uniqid === nextParticipant.forceUniqid)
+    if (forceIndex !== -1) {
+      // get all roles for current force and generate options for select
+      roleOptions = forces[forceIndex].roles.map((role): Option => ({
+        name: role.name,
+        uniqid: role.name,
+        value: role
+      }))
+      // selected value value for select
+      forceSelected = [forceIndex]
+    }
+  }
+
+  // get selected roles
+  const activeRoles: Array<number> = nextParticipant.roles.map(role => {
+    return roleOptions.findIndex(option => option.value.roleId === role)
+  }).filter(active => active !== -1)
+
+  // if this is a collaborative editing channel then the participant
```

Comment:
if this helper is just for collab channels, how about changing the type of the `nextParticipant` parameter to `ParticipantCollab`?

Reply:
I think the function `rowToParticipantCollab` should return the `ParticipantCollab` instead of `Participant` as default 😄

Reply:
I think we just need to update the `rowToParticipantCollab` to return the `ParticipantCollab ` data type, then we don't need to do a cast.